### PR TITLE
shogun: import from homebrew/science

### PR DESCRIPTION
**Pull request by vigsterkr**
and upgrade to 6.1.3

### CODE REVIEW

**Comment by sjackman** [comment]
Migrate from Homebrew/science https://github.com/Homebrew/homebrew-science/pull/6487

**Comment by sjackman** [comment]
Good to merge?

**Comment by vigsterkr** [comment]
@sjackman imo yes

File: Formula/shogun.rb [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+  depends_on "nlopt"
+  depends_on "eigen"
+  depends_on "arpack"
+  depends_on "glpk"
+  depends_on "r" => :optional
+  depends_on "lua" => :optional
+  depends_on "octave" => :optional
+  depends_on "opencv" => :optional
+  depends_on :java => :optional
+  depends_on :python => :optional
```
Comment (ilovezfs) [inline Formula/shogun.rb]:
```
depends_on :python if MacOS.version <= :snow_leopard
```
and don't make it optional.

File: Formula/shogun.rb [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+  depends_on "lzo"
+  depends_on "snappy"
+  depends_on "hdf5"
+  depends_on "json-c"
+  depends_on "readline"
+  depends_on "nlopt"
+  depends_on "eigen"
+  depends_on "arpack"
+  depends_on "glpk"
+  depends_on "r" => :optional
```
Comment (ilovezfs) [inline Formula/shogun.rb]: please remove this line and the r option

File: Formula/shogun.rb [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+  depends_on "lua" => :optional
```
Comment (ilovezfs) [inline Formula/shogun.rb]: please remove this line and the lua option

File: Formula/shogun.rb [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+  depends_on "octave" => :optional
```
Comment (ilovezfs) [inline Formula/shogun.rb]: please remove this line and the octave option

File: Formula/shogun.rb [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+  depends_on "opencv" => :optional
```
Comment (ilovezfs) [inline Formula/shogun.rb]: please remove this line and the opencv option

File: Formula/shogun.rb [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+  depends_on :java => :optional
```
Comment (ilovezfs) [inline Formula/shogun.rb]: please remove this line and the java option

File: Formula/shogun.rb [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+  depends_on :python => :optional
+
+  if build.with? "python"
+    include Language::Python::Virtualenv
```
Comment (ilovezfs) [inline Formula/shogun.rb]: please don't use the Virtualenv DSL. Instead use ` libexec/"vendor/lib/python2.7/site-packages"` as in the `aws-shell` formula

File: Formula/shogun.rb [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+
+    resource "matplotlib" do
+      url "https://files.pythonhosted.org/packages/15/89/240b4ebcd63bcdde9aa522fbd2e13f0af3347bea443cb8ad111e3b4c6f3a/matplotlib-1.5.2.tar.gz"
+      sha256 "8875d763c9e0d0ae01fefd5ebbe2b22bde5f080037f9467126d5dbee31785913"
+    end
+  end
+
+  needs :cxx11
+
+  def cmake_use(opt, arg)
```
Comment (ilovezfs) [inline Formula/shogun.rb]: please do not use functions like this in formulae

File: Formula/shogun.rb [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+      "-DLIB_INSTALL_DIR=#{lib}",
+      cmake_use("opencv", "OpenCV"),
+      cmake_use("r", "INTERFACE_R"),
+      cmake_use("octave", "INTERFACE_OCTAVE"),
+      cmake_use("lua", "INTERFACE_LUA"),
+      cmake_use("python", "INTERFACE_PYTHON"),
+      cmake_use("java", "INTERFACE_JAVA"),
+    ]
+
+    virtualenv_install_with_resources if build.with? "python"
```
Comment (ilovezfs) [inline Formula/shogun.rb]: see comment above

File: Formula/shogun.rb [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+
+    mkdir "build" do
+      system "cmake", "..", *args
+      system "make", "GoogleMock"
```
Comment (ilovezfs) [inline Formula/shogun.rb]: we don't run `make test` or `make check` except for security software

File: Formula/shogun.rb [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+      system "make"
```
Comment (ilovezfs) [inline Formula/shogun.rb]: `make install` alone is probably sufficient

File: Formula/shogun.rb [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+  bottle do
+    sha256 "06aafb46d4cd4e53ebed6f8bb32e1279b5a8b5de9aae2359261cee9de87fc2ca" => :sierra
+    sha256 "e5dd5b447ef40488fb32fa36e2eb943975a6b28db8ae27859e39cb8a87a077a9" => :el_capitan
+    sha256 "b3d320d345da09b55ffb98a59f4fac3b2f6c2672fcbcda693fe0e7a9ee4795a3" => :yosemite
+    sha256 "dfbd03cba5e1a134a520d6c06aceaa3c5143cad638fc208150d980a44e5252cf" => :x86_64_linux
+  end
+
+  depends_on "pkg-config" => :build
+  depends_on "cmake" => :build
+  depends_on "xz"
```
Comment (ilovezfs) [inline Formula/shogun.rb]: non-build deps should be alphabetized amongst themselves

File: Formula/shogun.rb [hunk 13]
```diff
@@ -0,0 +1,115 @@
...
+
+  bottle do
+    sha256 "06aafb46d4cd4e53ebed6f8bb32e1279b5a8b5de9aae2359261cee9de87fc2ca" => :sierra
+    sha256 "e5dd5b447ef40488fb32fa36e2eb943975a6b28db8ae27859e39cb8a87a077a9" => :el_capitan
+    sha256 "b3d320d345da09b55ffb98a59f4fac3b2f6c2672fcbcda693fe0e7a9ee4795a3" => :yosemite
+    sha256 "dfbd03cba5e1a134a520d6c06aceaa3c5143cad638fc208150d980a44e5252cf" => :x86_64_linux
+  end
+
+  depends_on "pkg-config" => :build
+  depends_on "cmake" => :build
```
Comment (ilovezfs) [inline Formula/shogun.rb]: build deps should be alphabetized amongst themselves

File: Formula/shogun.rb [hunk 14]
```diff
@@ -0,0 +1,115 @@
+class Shogun < Formula
+  desc "Large scale machine learning toolbox"
+  homepage "http://www.shogun-toolbox.org"
```
Comment (ilovezfs) [inline Formula/shogun.rb]: needs a trailing slash

**Review by ilovezfs — CHANGES_REQUESTED** [review]

File: Formula/shogun.rb (hunk 2 above)
Comment (vigsterkr) [reply Formula/shogun.rb]: @ilovezfs why?

File: Formula/shogun.rb (hunk 3 above)
Comment (vigsterkr) [reply Formula/shogun.rb]: @ilovezfs why?

File: Formula/shogun.rb (hunk 4 above)
Comment (vigsterkr) [reply Formula/shogun.rb]: @ilovezfs why?

File: Formula/shogun.rb (hunk 5 above)
Comment (vigsterkr) [reply Formula/shogun.rb]: @ilovezfs why?

File: Formula/shogun.rb (hunk 6 above)
Comment (vigsterkr) [reply Formula/shogun.rb]: @ilovezfs why?

File: Formula/shogun.rb (hunk 1 above)
Comment (vigsterkr) [reply Formula/shogun.rb]: why?

**Comment by ilovezfs** [comment]
We rarely allow formula options for new formulae or the migrations, especially when they're not broadly used.

```
Josephs-MacBook-Pro:~ joe$ brew formula-analytics shogun --days-ago=90
install events in the last 90 days for shogun
=====================================================================================================
1 | homebrew/science/shogun                                                            | 79 |  95.18%
2 | homebrew/science/shogun --with-python                                              |  2 |   2.41%
3 | homebrew/science/shogun --with-opencv                                              |  1 |   1.20%
4 | homebrew/science/shogun --with-r                                                   |  1 |   1.20%
=====================================================================================================
Total                                                                                  | 83 |    100%
=====================================================================================================

install_on_request events
=====================================================================================================
1 | homebrew/science/shogun                                                            | 76 |  96.20%
2 | homebrew/science/shogun --with-opencv                                              |  1 |   1.27%
3 | homebrew/science/shogun --with-python                                              |  1 |   1.27%
4 | homebrew/science/shogun --with-r                                                   |  1 |   1.27%
=====================================================================================================
Total                                                                                  | 79 |    100%
=====================================================================================================

BuildError events
=====================================================================================================
1 | homebrew/science/shogun                                                            | 28 |  96.55%
2 | homebrew/science/shogun --with-python                                              |  1 |   3.45%
=====================================================================================================
Total                                                                                  | 29 |    100%
=====================================================================================================
Josephs-MacBook-Pro:~ joe$ 
```

**Comment by vigsterkr** [comment]
@ilovezfs and how do you suppose those users would be able to use shogun's R, java, lua etc interface? download the src and compile themselves? i dont understand why that's a better option than actually brew handling it out of box..

**Comment by vigsterkr** [comment]
i dont see the clear reletionship between an option being enabled and how much it has been used in the last 90... i mean the purpose of a packaging system is to hide all the intricacies of compiling a software package by hand and give your users the convenience that they get a software with only one command line. now this gets taken away, which btw was until now part of the package since the package is part of brew?

**Comment by ilovezfs** [comment]
Correct, or keep a copy in their own tap. I'm open to making some of those required instead of optional if you feel it makes the formula more useful.

Options require source builds, which lead to build failures, and issue reports. Options that aren't broadly used sit like little time bombs waiting to go off when someone actually tries to use them and finds they've actually been broken for a while without out anyone noticing. We don't have the CI resources to test every combination so that's an inevitability.

**Comment by ilovezfs** [comment]
@vigsterkr the dependencies aren't really too much of a concern.

```
Josephs-MacBook-Pro:~ joe$ brew deps opencv
eigen
ffmpeg
ilmbase
jpeg
lame
libpng
libtiff
numpy
openexr
python
python3
tbb
x264
xvid
Josephs-MacBook-Pro:~ joe$ brew formula-analytics opencv
install events in the last 30 days for opencv
=====================================================================================================
1 | opencv                                                                         | 19,134 |  99.92%
2 | opencv --with-java                                                             |      6 |   0.03%
3 | opencv --with-contrib                                                          |      4 |   0.02%
4 | opencv --with-contrib --with-java                                              |      4 |   0.02%
5 | opencv --c++11                                                                 |      1 |   0.01%
6 | opencv --HEAD                                                                  |      1 |   0.01%
=====================================================================================================
Total                                                                              | 19,150 |    100%
=====================================================================================================

install_on_request events
=====================================================================================================
1 | opencv                                                                         | 17,642 |  99.91%
2 | opencv --with-java                                                             |      6 |   0.03%
3 | opencv --with-contrib                                                          |      4 |   0.02%
4 | opencv --with-contrib --with-java                                              |      4 |   0.02%
5 | opencv --c++11                                                                 |      1 |   0.01%
6 | opencv --HEAD                                                                  |      1 |   0.01%
=====================================================================================================
Total                                                                              | 17,658 |    100%
=====================================================================================================

BuildError events
=====================================================================================================
1 | opencv                                                                            | 233 |  99.57%
2 | opencv --with-contrib --with-java                                                 |   1 |   0.43%
=====================================================================================================
Total                                                                                 | 234 |    100%
=====================================================================================================
Josephs-MacBook-Pro:~ joe$ 
```
The formula has no options and we've had zero complaints about the number of deps (just a few people asking for java and qt to be *added*) and zero issue reports. Contrast with the situation in science where opencv/opencv3 were the most prolific generators of issue reports and the formulae had five million options.

**Comment by ilovezfs** [comment]
Octave is an even more extreme example:
```
Josephs-MacBook-Pro:~ joe$ brew deps octave
arpack
epstool
fftw
fig2dev
flac
fltk
fontconfig
freetype
gcc
gd
ghostscript
gl2ps
glpk
gmp
gnuplot
graphicsmagick
hdf5
imagemagick
jpeg
libogg
libpng
libsndfile
libtiff
libtool
libvorbis
little-cms2
lua
metis
pcre
plotutils
portaudio
pstoedit
qhull
qrupdate
readline
suite-sparse
szip
veclibfort
webp
xz
Josephs-MacBook-Pro:~ joe$ brew formula-analytics octave
install events in the last 30 days for octave
=====================================================================================================
1 | octave                                                                          | 9,040 |  97.88%
2 | octave --with-java                                                              |    98 |   1.06%
3 | octave --HEAD --with-java                                                       |    81 |   0.88%
4 | octave --HEAD                                                                   |    17 |   0.18%
=====================================================================================================
Total                                                                               | 9,236 |    100%
=====================================================================================================

install_on_request events
=====================================================================================================
1 | octave                                                                          | 7,300 |  97.44%
2 | octave --with-java                                                              |    95 |   1.27%
3 | octave --HEAD --with-java                                                       |    80 |   1.07%
4 | octave --HEAD                                                                   |    17 |   0.23%
=====================================================================================================
Total                                                                               | 7,492 |    100%
=====================================================================================================

BuildError events
=====================================================================================================
1 | octave --HEAD --with-java                                                          | 35 |  50.72%
2 | octave                                                                             | 24 |  34.78%
3 | octave --HEAD                                                                      |  7 |  10.14%
4 | octave --with-java                                                                 |  3 |   4.35%
=====================================================================================================
Total                                                                                  | 69 |    100%
=====================================================================================================
Josephs-MacBook-Pro:~ joe$ 
```
No one complains about the deps.

**Comment by ilovezfs** [comment]
> but ok if you think that it's better that i change all the optional flags to compulsory i'm fine with that as well.

I don't think we need to require them or add the options. The analytics really don't support either of those.

```
Josephs-MacBook-Pro:~ joe$ brew formula-analytics shogun --days-ago=180
install events in the last 180 days for shogun
=====================================================================================================
 1 | homebrew/science/shogun                                                          | 140 |  84.85%
 2 | homebrew/science/shogun --with-java                                              |   9 |   5.45%
 3 | homebrew/science/shogun --with-python                                            |   9 |   5.45%
 4 | homebrew/science/shogun --with-java --with-python                                |   1 |   0.61%
 5 | homebrew/science/shogun --with-lua --with-java --with-python                     |   1 |   0.61%
 6 | homebrew/science/shogun --with-opencv                                            |   1 |   0.61%
 7 | homebrew/science/shogun --with-r                                                 |   1 |   0.61%
 8 | homebrew/science/shogun --with-r --with-lua --with-java                          |   1 |   0.61%
 9 | homebrew/science/shogun --with-r --with-lua --with-java --with-python            |   1 |   0.61%
10 | homebrew/science/shogun --with-r --with-opencv --with-python                     |   1 |   0.61%
=====================================================================================================
Total                                                                                 | 165 |    100%
=====================================================================================================

install_on_request events
=====================================================================================================
 1 | homebrew/science/shogun                                                          | 130 |  84.42%
 2 | homebrew/science/shogun --with-java                                              |   9 |   5.84%
 3 | homebrew/science/shogun --with-python                                            |   8 |   5.19%
 4 | homebrew/science/shogun --with-java --with-python                                |   1 |   0.65%
 5 | homebrew/science/shogun --with-lua --with-java --with-python                     |   1 |   0.65%
 6 | homebrew/science/shogun --with-opencv                                            |   1 |   0.65%
 7 | homebrew/science/shogun --with-r                                                 |   1 |   0.65%
 8 | homebrew/science/shogun --with-r --with-lua --with-java                          |   1 |   0.65%
 9 | homebrew/science/shogun --with-r --with-lua --with-java --with-python            |   1 |   0.65%
10 | homebrew/science/shogun --with-r --with-opencv --with-python                     |   1 |   0.65%
=====================================================================================================
Total                                                                                 | 154 |    100%
=====================================================================================================

BuildError events
=====================================================================================================
1 | homebrew/science/shogun                                                            | 31 |  68.89%
2 | homebrew/science/shogun --with-python                                              |  6 |  13.33%
3 | homebrew/science/shogun --with-java                                                |  3 |   6.67%
4 | homebrew/science/shogun --with-java --with-python                                  |  1 |   2.22%
5 | homebrew/science/shogun --with-lua --with-java --with-python                       |  1 |   2.22%
6 | homebrew/science/shogun --with-r --with-lua --with-java                            |  1 |   2.22%
7 | homebrew/science/shogun --with-r --with-lua --with-java --with-python              |  1 |   2.22%
8 | homebrew/science/shogun --with-r --with-opencv --with-python                       |  1 |   2.22%
=====================================================================================================
Total                                                                                  | 45 |    100%
=====================================================================================================
Josephs-MacBook-Pro:~ joe$ brew formula-analytics shogun --days-ago=90
install events in the last 90 days for shogun
=====================================================================================================
1 | homebrew/science/shogun                                                            | 79 |  95.18%
2 | homebrew/science/shogun --with-python                                              |  2 |   2.41%
3 | homebrew/science/shogun --with-opencv                                              |  1 |   1.20%
4 | homebrew/science/shogun --with-r                                                   |  1 |   1.20%
=====================================================================================================
Total                                                                                  | 83 |    100%
=====================================================================================================

install_on_request events
=====================================================================================================
1 | homebrew/science/shogun                                                            | 76 |  96.20%
2 | homebrew/science/shogun --with-opencv                                              |  1 |   1.27%
3 | homebrew/science/shogun --with-python                                              |  1 |   1.27%
4 | homebrew/science/shogun --with-r                                                   |  1 |   1.27%
=====================================================================================================
Total                                                                                  | 79 |    100%
=====================================================================================================

BuildError events
=====================================================================================================
1 | homebrew/science/shogun                                                            | 28 |  96.55%
2 | homebrew/science/shogun --with-python                                              |  1 |   3.45%
=====================================================================================================
Total                                                                                  | 29 |    100%
=====================================================================================================
Josephs-MacBook-Pro:~ joe$ brew formula-analytics shogun --days-ago=30
install events in the last 30 days for shogun
=====================================================================================================
1 | homebrew/science/shogun                                                            | 23 |  95.83%
2 | homebrew/science/shogun --with-r                                                   |  1 |   4.17%
=====================================================================================================
Total                                                                                  | 24 |    100%
=====================================================================================================

install_on_request events
=====================================================================================================
1 | homebrew/science/shogun                                                            | 21 |  95.45%
2 | homebrew/science/shogun --with-r                                                   |  1 |   4.55%
=====================================================================================================
Total                                                                                  | 22 |    100%
=====================================================================================================

BuildError events
=====================================================================================================
1 | homebrew/science/shogun                                                            | 15 | 100.00%
=====================================================================================================
Josephs-MacBook-Pro:~ joe$ brew formula-analytics shogun --days-ago=15
install events in the last 15 days for shogun
=====================================================================================================
1 | homebrew/science/shogun                                                            | 19 |  95.00%
2 | homebrew/science/shogun --with-r                                                   |  1 |   5.00%
=====================================================================================================
Total                                                                                  | 20 |    100%
=====================================================================================================

install_on_request events
=====================================================================================================
1 | homebrew/science/shogun                                                            | 18 |  94.74%
2 | homebrew/science/shogun --with-r                                                   |  1 |   5.26%
=====================================================================================================
Total                                                                                  | 19 |    100%
=====================================================================================================

BuildError events
=====================================================================================================
1 | homebrew/science/shogun                                                            | 12 | 100.00%
=====================================================================================================
Josephs-MacBook-Pro:~ joe$ 
```

**Comment by vigsterkr** [comment]
@ilovezfs can a core package depend on another tap's package, i.e. homebrew/science/jblas ?

**Comment by ilovezfs** [comment]
Nope, that's not permissible. So `jblas` would need to be vendored into the formula if it's needed.

**Comment by vigsterkr** [comment]
it's required for the java interface

**Comment by ilovezfs** [comment]
```
class Jblas < Formula
  homepage "https://mikiobraun.github.io/jblas"
  url "https://mikiobraun.github.io/jblas/jars/jblas-1.2.3.jar"
  sha256 "e9328d4e96db6b839abf50d72f63626b2309f207f35d0858724a6635742b8398"

  def install
    (share / "java").install "jblas-#{version}.jar" => "jblas.jar"
  end
end
```

So that could go in a resource block.

**Comment by vigsterkr** [comment]
@ilovezfs thnx. is there a way to let `lapack` dependency in a package? as the audit complains:
```Dependency 'lapack' may be unnecessary as it is provided by macOS; try to build this formula without it.```

note it'd be good to have lapacke.dylib, that is not available on the system itself

**Comment by ilovezfs** [comment]
We can't use Accelerate.framework?

**Comment by vigsterkr** [comment]
@ilovezfs it is using Accelerate.framekwork but for better performance it'd be good to have lapacke.dylib see eigen's manual:
https://eigen.tuxfamily.org/dox/TopicUsingBlasLapack.html

**Comment by ilovezfs** [comment]
@vigsterkr sure that's fine. It'll fail `--new-formula` audit but not subsequent audits.

**Comment by vigsterkr** [comment]
cool. so i followed the python part in `aws-shell` formula but currently i'm getting this error:
```Last 15 lines from /Users/wiking/Library/Logs/Homebrew/shogun/02.python:
make arguments: install
Verifying Shogun preconditions
Shogun build environment completed tasks: cmake: [True] -  make: [True] - make install: [True]
Setting Shogun Python packages location
Traceback (most recent call last):
  File "<string>", line 4, in <module>
  File "setup.py", line 299, in <module>
    version = get_shogun_version(),
  File "setup.py", line 205, in get_shogun_version
    if shogun_preconditions():
  File "setup.py", line 114, in shogun_preconditions
    shogun_bootstrap()
  File "setup.py", line 91, in shogun_bootstrap
    shogun_python_packages_location = glob.glob(os.path.join(shogun_generated_install, 'lib/*/dist-packages'))[0]
IndexError: list index out of range
/usr/local/Homebrew/Library/Homebrew/build.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/python.rb
/usr/local/Homebrew/Library/Homebrew/debrew.rb:11:in `raise'
BuildError: Failed executing: python -c import\ setuptools,\ tokenize
__file__\ =\ 'setup.py'
exec(compile(getattr(tokenize,\ 'open',\ open)(__file__).read()
\ \ .replace('\r\n',\ '\n'),\ __file__,\ 'exec'))
 --no-user-cfg install --prefix=/usr/local/Cellar/shogun/6.1.3/libexec --single-version-externally-managed --record=installed.txt```


the relevant part of the new formula:
```
    ENV["PYTHONPATH"] = libexec/"lib/python2.7/site-packages"
    ENV.prepend_create_path "PYTHONPATH", libexec/"vendor/lib/python2.7/site-packages"

    res = %w[numpy]
    res.each do |r|
      resource(r).stage do
        system "python", *Language::Python.setup_install_args(libexec/"vendor")
      end
    end

    system "python", *Language::Python.setup_install_args(libexec)
    bin.install Dir[libexec/"bin/*"]
    env = {
      :PATH => "#{libexec}/vendor/bin:$PATH",
      :PYTHONPATH => ENV["PYTHONPATH"],
      :LAPACKE_PATH => '#{Formula["lapack"].opt_lib}',
    }
    bin.env_script_all_files(libexec/"bin", env)
```

any ideas?

**Comment by vigsterkr** [comment]
@ilovezfs just pushed the failing changes, lemme know if you know the solution to this problem

**Comment by ilovezfs** [comment]
```
    shogun_python_packages_location = glob.glob(os.path.join(shogun_generated_install, 'lib/*/dist-packages'))[0]
IndexError: list index out of range
```
That suggests to me that the glob is empty.

**Comment by ilovezfs** [comment]
You'll also need to add `-DJBLAS=#{libexec}/jblas-1.2.3.jar`

**Comment by ilovezfs** [comment]
I'm getting this build failure
```
[  2%] Building CXX object src/shogun/CMakeFiles/libshogun.dir/classifier/svm/NewtonSVM.cpp.o
cd /tmp/shogun-20171231-72538-9cb84g/shogun-6.1.3/build/src/shogun && /usr/local/Homebrew/Library/Homebrew/shims/super/clang++   -I/tmp/shogun-20171231-72538-9cb84g/shogun-6.1.3/src -I/tmp/shogun-20171231-72538-9cb84g/shogun-6.1.3/build/src -I/tmp/shogun-20171231-72538-9cb84g/shogun-6.1.3/src/gpl -isystem /tmp/shogun-20171231-72538-9cb84g/shogun-6.1.3/build/src/shogun/lib/external -isystem /usr/local/include/eigen3 -I/usr/local/include -I/usr/local/lib/../include -I/usr/local/include/json-c -I/usr/include/libxml2 -I/usr/local/Cellar/hdf5/1.10.1_2/include -I/usr/local/opt/szip/include  -Wall -Wno-unused-parameter -Wformat -Wformat-security -Wparentheses -Wshadow -Wno-unknown-pragmas -Wno-deprecated  -O3 -funroll-loops -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk -mmacosx-version-min=10.11 -fPIC   -std=c++11 -o CMakeFiles/libshogun.dir/classifier/svm/NewtonSVM.cpp.o -c /tmp/shogun-20171231-72538-9cb84g/shogun-6.1.3/src/shogun/classifier/svm/NewtonSVM.cpp
In file included from /tmp/shogun-20171231-72538-9cb84g/shogun-6.1.3/src/shogun/classifier/svm/NewtonSVM.cpp:20:
/tmp/shogun-20171231-72538-9cb84g/shogun-6.1.3/src/shogun/mathematics/lapack.h:38:10: fatal error: 'vecLib/cblas.h' file not found
#include <vecLib/cblas.h>
         ^
1 error generated.
make[2]: *** [src/shogun/CMakeFiles/libshogun.dir/classifier/svm/NewtonSVM.cpp.o] Error 1
make[1]: *** [src/shogun/CMakeFiles/libshogun.dir/all] Error 2
make: *** [all] Error 2
bash-3.2$ 
```

**Pull request closed without merging**